virtual_channels/contracts/Splitter.sol

```diff
@@ -12,28 +12,52 @@ contract Splitter {
   */
   address payable abAddress;
   address payable intermediaryAddress;
+  address payable aAddress;
+  address payable bAddress;
   uint256 collateral;
 
   // This variable is set after the first thing is received
   uint256 received;
+  uint256 refundsAt;
 
   constructor (
-    address payable[2] memory addresses,
+    address payable[4] memory addresses,
     uint256 _collateral
   ) public {
     abAddress = addresses[0];
     intermediaryAddress = addresses[1];
+    aAddress = addresses[2];
+    bAddress = addresses[3];
     collateral = _collateral;
   }
 
   function deposit() public payable {
+    require(refundsAt > 0, "must call becomeSplitter first");
     require(msg.value == collateral, "Received unexpected msg.value");
-    if (received == 0) {
-      received = msg.value;
-      abAddress.transfer(collateral);
-    } else {
-      intermediaryAddress.transfer(collateral);
+
+    if (refundsAt == 1) /* is splitter */ {
+      if (received == 0) {
+        received = msg.value;
+        abAddress.transfer(collateral);
+      } else {
+        intermediaryAddress.transfer(collateral);
+      }
+    } else /* is refunder */ {
+      aAddress.transfer(collateral / 2);
+      bAddress.transfer(collateral / 2);
     }
+
+  }
+
+  function startBecomeSomething() public {
+    require(refundsAt == 0, "startExit already called");
+    refundsAt = block.number + 1000;
```

Comment:
Would we want to pass `1000` from the constructor? This will makes testing far easier since we can pass a smaller number.